packages/salesforcedx-utils-vscode/src/commands/sfdxCommandlet.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as vscode from 'vscode';
+import { CliCommandExecutor, Command, CommandExecution } from '../cli';
+import {
+  Measurements,
+  Properties,
+  TelemetryData,
+  TelemetryService
+} from '../index';
+import {
+  ContinueResponse,
+  ParametersGatherer,
+  PostconditionChecker,
+  PreconditionChecker
+} from '../types';
+import { getRootWorkspacePath } from '../workspaces';
+// import { ChannelService } from './channelService';
+import { notificationService, ProgressNotification } from './index';
+import { EmptyPostChecker } from './postconditionCheckers';
+
+export interface FlagParameter<T> {
+  flag: T;
+}
+
+export interface CommandletExecutor<T> {
+  execute(response: ContinueResponse<T>): void;
+  readonly onDidFinishExecution?: vscode.Event<[number, number]>;
+}
+
+export abstract class SfdxCommandletExecutor<T>
+  implements CommandletExecutor<T> {
+  protected showChannelOutput = true;
+  protected executionCwd = getRootWorkspacePath();
+  protected onDidFinishExecutionEventEmitter = new vscode.EventEmitter<
+    [number, number]
+  >();
+  public readonly onDidFinishExecution: vscode.Event<[number, number]> = this
+    .onDidFinishExecutionEventEmitter.event;
+
+  protected attachExecution(
+    execution: CommandExecution,
+    cancellationTokenSource: vscode.CancellationTokenSource,
+    cancellationToken: vscode.CancellationToken
+  ) {
+    /* channelService.streamCommandOutput(execution);
+
+    if (this.showChannelOutput) {
+      channelService.showChannelOutput();
+    } */
```

Comment:
are we planning on moving the channelService and adding in this piece in a future PR?